src/lms5xx_node.cpp

```diff
@@ -0,0 +1,263 @@
+#include <csignal>
+#include <cstdio>
+#include <LMS1xx/lms5xx.h>
+#include <ros/ros.h>
+#include <sensor_msgs/LaserScan.h>
+#include <sensor_msgs/MultiEchoLaserScan.h>
+
+#define DEG2RAD M_PI/180.0
+
+void usage()
+{
+  std::cout << "LMS5xx_node" << std::endl;
+  std::cout << "LMS5xx series laser scanner node." << std::endl;
+  std::cout << "If configured for all echoes, a MultiEchoLaserScan message will be published"
+            " on the \"multi_echo\" topic in addition to the regular, single echo \"scan\" topic." << std::endl << std::endl;
+  std::cout << "Parameters:" << std::endl;
+  std::cout << "    host      The IP of the scanner" << std::endl;
+  std::cout << "    port      The port to connect on" << std::endl;
+  std::cout << "    frame_id  Frame id of the laser, defaults to \"laser\"." << std::endl;
+  std::cout << "    echoes    One of \"first\", \"last\" or \"all\"." << std::endl;
+}
+
+bool setup(LMS5xx &laser, sensor_msgs::LaserScan &scan_msg, sensor_msgs::MultiEchoLaserScan &multi_scan_msg,
+           const LMS5xx::EchoFilter echo_mode)
+{
+  ScanConfig cfg;
+  ScanOutputRange output_range;
+  ScanDataConfig dataCfg;
+
+  ROS_DEBUG("Logging in to laser.");
+  laser.login();
+  cfg = laser.get_scan_config();
+  output_range = laser.get_scan_output_range();
+
+  if (cfg.scan_frequency != 5000)
+  {
+    laser.disconnect();
+    ROS_WARN("Unable to get laser output range. Retrying.");
+    ros::Duration(1).sleep();
+    return false;
+  }
+
+  ROS_INFO("Connected to laser.");
+
+  ROS_DEBUG("Laser configuration: scaningFrequency %d, activeSensors %d, angleResolution %d, startAngle %d, stopAngle %d",
+           cfg.scan_frequency, cfg.num_sectors, cfg.angualar_resolution, cfg.start_angle, cfg.stop_angle);
+  ROS_DEBUG("Laser output range: angleResolution %d, startAngle %d, stopAngle %d",
+           output_range.angular_resolution, output_range.start_angle, output_range.stop_angle);
+
+  scan_msg.range_min = 0.01;
+  scan_msg.range_max = 20.0; // TODO: value?
```

Comment:
This is higher! Check the spec sheet and adjust :-) Alternative is to pass this in via a parameter. This allows you to also clip  the values into a reasonable range for your application.